pkg/secrets/pushtofile/secret_group.go

```diff
@@ -75,24 +77,33 @@ func (sg *SecretGroup) pushToFileWithDeps(
 		sg.SecretSpecs,
 	)
 	if err != nil {
-		return err
+		return
 	}
 
 	//// Open and push to file
 	wc, err := depOpenWriteCloser(sg.FilePath, sg.FilePermissions)
 	if err != nil {
-		return err
+		return
 	}
+
 	defer func() {
 		_ = wc.Close()
+		if r := recover(); r != nil {
+			err = fmt.Errorf("panic recovered while executing template with secret values")
```

Comment:
More context in the message e.g.
```
failed to execute template, with secret values, on push to file for secret group %q
```

Reply:
I used this example as a general template execution failure message, whether recovering from panic or returning an error.